build.proj

```diff
@@ -46,32 +47,36 @@
     <Message Text="Dont remove this target" />
   </Target>
 
-  <Target Name="BuildDotnetCliBuildFramework" 
-          Inputs="@(DotnetCliBuildFrameworkInputs)" 
+  <Target Name="BuildDotnetCliBuildFramework"
+          Inputs="@(DotnetCliBuildFrameworkInputs)"
           Outputs="$(CLIBuildDll)"
           DependsOnTargets="MSBuildWorkaroundTarget">
     <PropertyGroup>
       <DotnetCliBuildDirectory>$(MSBuildThisFileDirectory)/build_projects/dotnet-cli-build</DotnetCliBuildDirectory>
     </PropertyGroup>
 
+    <ItemGroup>
+      <DotnetSdkDirectories Include="$([System.IO.Directory]::GetDirectories(&quot;$(Stage0Path)/sdk&quot;))" />
+      <AzureStorageToCopy Include="@(DotnetSdkDirectories)">
+        <Source>$(DotnetCliBuildDirectory)/bin/Microsoft.WindowsAzure.Storage.dll</Source>
+        <Destination>%(Identity)/Microsoft.WindowsAzure.Storage.dll</Destination>
+      </AzureStorageToCopy>
+    </ItemGroup>
+
     <Exec Command="$(DotnetStage0) restore" WorkingDirectory="$(DotnetCliBuildDirectory)"/>
 
-    <Exec Command="$(DotnetStage0) publish -o $(DotnetCliBuildDirectory)/bin --framework netcoreapp1.0" WorkingDirectory="$(DotnetCliBuildDirectory)"/>    
+    <Exec Command="$(DotnetStage0) publish -o $(DotnetCliBuildDirectory)/bin --framework netcoreapp1.0" WorkingDirectory="$(DotnetCliBuildDirectory)"/>
+
+    <!-- This is a work around to issue https://github.com/Microsoft/msbuild/issues/658 -->
+    <Copy SourceFiles="%(AzureStorageToCopy.Source)"
```

Comment:
Then I will have to go add that target in a bunch of different places, because it would have to run after this. Unless I used the RunsAfter or whatever that's called, which we have said in the past is not really something we want to use.